autoscaler/nanny/scaler_parser.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nanny
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+type replicaParams map[string]int
+
+type scalerParams struct {
+	CoresToReplicasMap replicaParams `json:"cores_to_replicas_map"`
+}
+
+// parseScalerParams Parse the scaler params JSON string
+func parseScalerParams(data []byte) (params *scalerParams, err error) {
+	var p scalerParams
+	if err := json.Unmarshal(data, &p); err != nil {
+		return nil, fmt.Errorf("Could not parse scalar parameters (%s)", err)
+	}
+	for k, v := range p.CoresToReplicasMap {
+		k64, err := strconv.ParseInt(k, 10, 32)
```

Comment:
why parse it here, then discard it and then parse it again later?  If you keep this struct, just parse once and save it as a map[int]int.